game/gameboard.py

```diff
@@ -149,7 +146,7 @@ def set_board_from_matrix(self, matrix):
             a boolean value
         """
         numpy_matrix = numpy.asarray(matrix)
-        if self._validate_matrix(numpy_matrix):
```

Comment:
I wanted to write some tests for this method, but figured out that the python linter complained about me trying to call private methods on the test, so I made it public for a while and then I think I forgot to return it to its previous condition. Is that OK if I want to write tests for private methods? (It was really helpful) I've read that it's supposed to be tested indirectly from the non-private methods' tests